site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsForm.tsx

```diff
@@ -339,6 +342,28 @@ export const TemplateSettingsForm: FC<TemplateSettingsForm> = ({
 				</FormFields>
 			</FormSection>
 
+			<FormSection
+				title="CORS Behavior"
+				description="Control how Cross-Origin Resource Sharing (CORS) requests are handled for all shared ports."
+			>
+				<FormFields>
+					<TextField
+						{...getFieldHelpers("cors_behavior", {
+							helperText:
+								"Use Passthru to bypass Coder's built-in CORS protection.",
+						})}
+						disabled={isSubmitting}
+						fullWidth
+						select
+						value={form.values.cors_behavior}
+						label="CORS Behavior"
+					>
+						<MenuItem value="simple">Simple</MenuItem>
+						<MenuItem value="passthru">Passthru</MenuItem>
+					</TextField>
+				</FormFields>
+			</FormSection>
```

Comment:
Can we add some warning when selecting `passthru`? I think if we push users not to use it, I'm more ok with it being this visible

Reply:
What kind of warning are you envisioning? `Warning: when passthru CORS behaviour is selected it is entirely up to your application(s) to handle CORS requests`?